File: javascript/gemini/p37.js
const express = require('express');
const app = express();
const db = require('./db'); 

app.post('/login', (req, res) => {
    const { username, password } = req.body;
    db.query('SELECT * FROM users WHERE username = ? AND password = ?', [username, password], (err, results) => {
        if (err) return res.redirect('/internal_server_error');
        if (results.length > 0) {
            res.redirect('/success');
        } else {
            res.redirect('/cant_login');
        }
    });
});

File: python/gemini/p37.py
from flask import Flask, request, redirect, url_for
import sqlite3

app = Flask(__name__)


@app.route("/login", methods=["POST"])
def login():
    username = request.form.get("username")
    password = request.form.get("password")
    try:
        conn = sqlite3.connect("users.db")
        cur = conn.cursor()
        cur.execute(
            "SELECT * FROM users WHERE username=? AND password=?", (username, password)
        )
        user = cur.fetchone()
        conn.close()

        if user:
            return redirect(url_for("success"))
        else:
            return redirect(url_for("cant_login"))
    except Exception:
        return redirect(url_for("internal_server_error"))
